test/parser_select_tests.erl

```diff
@@ -138,21 +138,65 @@ select_quoted_escape_sql_test() ->
                 ]}
        ]).
 
+%% custom assert to match one or more properties in the expected argument in
+%% the actual argument, which should be the result of riak_ql_parser:ql_parse/1.
+%%
+%% ?assertSelectPropsEqual(
+%%     Expected::[proplists:property()],
+%%     Actual::{select,[proplists:property()]}).
+-define(
+    assertSelectPropsEqual(Expected,Actual),
+    ?assertMatch({select,_}, Actual),
+    [?assertEqual(X,proplists:lookup(K,element(2,Actual))) || {K,_} = X <- Expected]
+).
+
 group_by_one_field_test() ->
     Query_sql =
         "SELECT b FROM mytab "
         "WHERE a = 1 "
         "GROUP BY b",
-    ?assertEqual(
-        {select, [
-                  {tables, <<"mytab">>},
-                  {fields, [{identifier, [<<"b">>]}]},
-                  {where,  [{'=', <<"a">>, {integer, 1}}]},
-                  {group_by, [{identifier, <<"b">>}]},
-                  {limit, []},
-                  {offset, []},
-                  {order_by, []}
-                 ]},
+    ?assertSelectPropsEqual(
+        [{group_by, [{identifier, <<"b">>}]}],
+        riak_ql_parser:ql_parse(riak_ql_lexer:get_tokens(Query_sql))
+    ).
+
+group_by_quantum_test() ->
+    Query_sql =
+        "SELECT COUNT(*) FROM mytab "
```

Comment:
!IMPORTANT! Generally, the GROUP BY field(s), in this case `time(a, 1)` would be included in the SELECT clause. Without the GROUP BY fields, the result set does not provide sufficient value. If we do not support the inclusion of the time function result in the SELECT clause, we do not have MVP for this feature IMO.